schema/tests/enum_with_various_discriminants.rs

```diff
@@ -28,18 +30,22 @@ fn discriminant() {
                 variants: vec![
                     EnumVariant {
                         tag: "A".to_owned(),
+                        discriminant: 1,
                         ty: None,
                     },
                     EnumVariant {
                         tag: "B".to_owned(),
+                        discriminant: 1,
```

Comment:
it's funny that with SCALE we can have two structs with the same discriminant

Reply:
one more reason why I dislike allowing discriminants

Reply:
Why wouldn't you add a guard into the proc macro to prevent such collision?

Reply:
Because there's still ways of implementing `Encode` and `Decode` with a collision, e.g. without using our macro, I think we also need a marker trait.  This will ensure that the user is shown an error message if they do something potentially buggy.